bcda/api/requests_test.go

```diff
@@ -123,159 +127,80 @@ func (s *RequestsTestSuite) TestRunoutDisabled() {
-		{"Invalid output format (x-custom)", reqParams{outputFormat: "x-custom"}, nil, "_outputFormat parameter must be application/fhir+ndjson, application/ndjson, or ndjson"},
-		{"Invalid query parameter (extra ?)", reqParams{}, map[string]string{"?_since": "2020-09-13T08:00:00.000-05:00"}, "Invalid parameter: query parameters cannot start with ?"},
-	}
-
-	for _, tt := range tests {
-		u, err := url.Parse("/api/v1")
-		if err != nil {
-			s.Failf("Failed to parse URL %s", err.Error())
-		}
-
-		q := u.Query()
-		if len(tt.reqParams.types) > 0 {
-			q.Set("_type", strings.Join(tt.reqParams.types, ","))
-		}
-		if len(tt.reqParams.since) > 0 {
-			q.Set("_since", tt.reqParams.since)
-		}
-		if len(tt.reqParams.outputFormat) > 0 {
-			q.Set("_outputFormat", tt.reqParams.outputFormat)
-		}
-		for key, value := range tt.extraQueryParams {
-			q.Set(key, value)
+// TestRequests verifies that we can initiate an export job for all resource types using all the different handlers
+func (s *RequestsTestSuite) TestRequests() {
+
+	h := newHandler([]string{"ExplanationOfBenefit", "Coverage", "Patient"}, "/v1/fhir", s.db)
+
+	// Use a mock to ensure that this test does not generate artifacts in the queue for other tests
+	enqueuer := &queueing.MockEnqueuer{}
+	enqueuer.On("AddJob", mock.Anything, mock.Anything).Return(nil)
+	h.Enq = enqueuer
+
+	// Test Group and Patient
+	// Patient, Coverage, and ExplanationOfBenefit
+	// with And without Since parameter
+	resources := []string{"Patient", "ExplanationOfBenefit", "Coverage"}
+	sinces := []time.Time{{}, time.Now().Round(time.Millisecond).Add(-24 * time.Hour)}
+	groupIDs := []string{"all", "runout"}
+
+	// Validate group requests
+	for _, resource := range resources {
+		for _, since := range sinces {
+			for _, groupID := range groupIDs {
+				rp := middleware.RequestParameters{
+					Version:       "v1",
+					ResourceTypes: []string{resource},
+					Since:         since,
+				}
+				rr := httptest.NewRecorder()
+				req := s.genGroupRequest(groupID, rp)
+				h.BulkGroupRequest(rr, req)
+				assert.Equal(s.T(), http.StatusAccepted, rr.Code)
+			}
 		}
-
-		u.RawQuery = q.Encode()
-		req := httptest.NewRequest("GET", u.String(), nil)
-		rctx := chi.NewRouteContext()
-		rctx.URLParams.Add("groupId", "all")
-		req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
-
-		s.T().Run(fmt.Sprintf("%s-group", tt.name), func(t *testing.T) {
-			rr := httptest.NewRecorder()
-			h.BulkGroupRequest(rr, req)
-			assert.Equal(t, http.StatusBadRequest, rr.Code)
-			assert.Contains(t, rr.Body.String(), tt.errMsg)
-		})
-
-		s.T().Run(fmt.Sprintf("%s-patient", tt.name), func(t *testing.T) {
-			rr := httptest.NewRecorder()
-			h.BulkGroupRequest(rr, req)
-			assert.Equal(t, http.StatusBadRequest, rr.Code)
-			assert.Contains(t, rr.Body.String(), tt.errMsg)
-		})
 	}
-}
 
-func (s *RequestsTestSuite) TestCheck429() {
```

Comment:
Replace by test in [ratelimit_test.go](https://github.com/CMSgov/bcda-app/pull/673/files#diff-75b0d1f2ed7de3581ea31fc1ebe5a2747d66a4e5f65fe8034494cc040b131d39R46)